lib/base.ts

```diff
@@ -76,7 +78,13 @@ export class TranspilerBase {
   maybeVisitTypeArguments(n: {typeArguments?: ts.NodeArray<ts.TypeNode>}) {
     if (n.typeArguments) {
       this.emit('<');
-      this.visitList(n.typeArguments);
+      this.visitList(n.typeArguments, ',', (node) => {
```

Comment:
As discussed offline, I think it'd be better to make this code less generic, and only specifically handle the case of a single type argument that's `void` by just leaving out the type argument.

This code is arguable better by being more generic, but in this case we should be restrictive - we shouldn't support patterns that are going to cause odd APIs for Dart users, when we really only need to support `Promise<void>` and `EventEmitter<void>`.